tests/unit/lib/charms/kubernetes_charm_libraries/v0/test_kubernetes_volumes_patch.py

```diff
@@ -0,0 +1,652 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+from unittest.mock import Mock, patch
+
+import httpx
+from charms.kubernetes_charm_libraries.v0.kubernetes_volumes_patch import (  # type: ignore[import]
+    K8sVolumePatchChangedCharmEvents,
+    KubernetesClient,
+    KubernetesVolumesPatchCharmLib,
+    RequestedVolume,
+)
+from lightkube.core.exceptions import ApiError
+from lightkube.models.apps_v1 import StatefulSet, StatefulSetSpec
+from lightkube.models.core_v1 import (
+    Container,
+    EmptyDirVolumeSource,
+    PodSpec,
+    PodTemplateSpec,
+    ResourceRequirements,
+    Volume,
+    VolumeMount,
+)
+from lightkube.models.meta_v1 import LabelSelector
+from lightkube.resources.apps_v1 import StatefulSet as StatefulSetResource
+from lightkube.resources.core_v1 import Pod
+from lightkube.types import PatchType
+from ops.charm import CharmBase
+from ops.testing import Harness
+
+VOLUMES_LIBRARY_PATH = "charms.kubernetes_charm_libraries.v0.kubernetes_volumes_patch"
+
+
+class TestKubernetes(unittest.TestCase):
+    @patch("lightkube.core.client.GenericSyncClient", new=Mock)
+    def setUp(self) -> None:
+        self.namespace = "whatever ns"
+        self.kubernetes_volumes = KubernetesClient(namespace=self.namespace)
+
+    @patch("lightkube.core.client.Client.patch")
+    def test_given_no_requested_volume_when_patch_statefulset_then_statefulset_is_not_patched(
+        self, patch_patch
+    ):
+        requested_volumes = []
+
+        self.kubernetes_volumes.patch_volumes(
+            statefulset_name="whatever statefulset name",
+            requested_volumes=requested_volumes,
+            container_name="container-name",
+        )
+
+        patch_patch.assert_not_called()
+
+    @patch("lightkube.core.client.Client.replace")
+    def test_given_no_requested_volume_when_replace_statefulset_then_statefulset_is_not_replaced(
+        self, patch_replace
+    ):
+        requested_volumes = []
+
+        self.kubernetes_volumes.remove_volumes(
+            statefulset_name="whatever statefulset name",
+            requested_volumes=requested_volumes,
+            container_name="container-name",
+        )
+
+        patch_replace.assert_not_called()
+
+    @patch("lightkube.core.client.Client.patch")
+    @patch("lightkube.core.client.Client.get")
+    def test_given_statefulset_doesnt_have_requested_volumes_when_patch_statefulset_then_statefulset_is_patched(  # noqa: E501
+        self, patch_get, patch_patch
+    ):
+        container_name = "whatever container name"
+        statefulset_name = "whatever statefulset name"
```

Comment:
As a minor improvement, we can define global variables for those values which are used in several tests.